horiz_interp/horiz_interp.F90

```diff
@@ -122,10 +123,18 @@ module horiz_interp_mod
 !!      interpolations. To reinitialize this variable for a different grid-to-grid
 !!      interpolation you must first use the "horiz_interp_del" interface.
  interface horiz_interp_new
-    module procedure horiz_interp_new_1d     ! Source grid is 1d, destination grid is 1d
-    module procedure horiz_interp_new_1d_src ! Source grid is 1d, destination grid is 2d
-    module procedure horiz_interp_new_2d     ! Source grid is 2d, destination grid is 2d
-    module procedure horiz_interp_new_1d_dst ! Source grid is 2d, destination grid is 1d
+    ! Source grid is 1d, destination grid is 1d
+    module procedure horiz_interp_new_1d_r4
+    module procedure horiz_interp_new_1d_r8
+    ! Source grid is 1d, destination grid is 2d
+    module procedure horiz_interp_new_1d_src_r4
+    module procedure horiz_interp_new_1d_src_r8
+    ! Source grid is 2d, destination grid is 2d
+    module procedure horiz_interp_new_2d_r4
+    module procedure horiz_interp_new_2d_r8
+    ! Source grid is 2d, destination grid is 1d
+    module procedure horiz_interp_new_1d_dst_r4
+    module procedure horiz_interp_new_1d_dst_r8
```

Comment:
I know you didn't do this, but these all have "new" in the name,

Reply:
@thomas-robinson do you think it's a good idea to change these names in a separate FMS PR that involves getting rid of all the "old" subroutines?

Reply:
I talked to @rem1776 about it and we will leave it because this is creating a new instance and not "newer" than the other routines.